File: eRent/eRent.Services/Database/Rent.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace eRent.Services.Database
{
    public class Rent
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int PropertyId { get; set; }

        [Required]
        public int UserId { get; set; } // FK to User (tenant)

        [Required]
        public DateTime StartDate { get; set; }

        [Required]
        public DateTime EndDate { get; set; }

        [Required]
        public bool IsDailyRental { get; set; } // true for daily, false for monthly

        [Required]
        public decimal TotalPrice { get; set; }

        [Required]
        public int RentStatusId { get; set; } = 1; // FK to RentStatus (default: Pending)

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.Now;

        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        [ForeignKey("PropertyId")]
        public Property Property { get; set; } = null!;

        [ForeignKey("UserId")]
        public User User { get; set; } = null!;

        [ForeignKey("RentStatusId")]
        public RentStatus RentStatus { get; set; } = null!;

        // Navigation property for ReviewRents
        public ICollection<ReviewRent> ReviewRents { get; set; } = new List<ReviewRent>();
    }
}

File: Startupba/UI/erent_desktop/lib/model/rent.dart
import 'package:json_annotation/json_annotation.dart';

part 'rent.g.dart';

@JsonSerializable()
class Rent {
  final int id;
  final int propertyId;
  final String propertyTitle;
  final int userId;
  final String userName;
  final DateTime startDate;
  final DateTime endDate;
  final bool isDailyRental;
  final double totalPrice;
  final int rentStatusId;
  final String rentStatusName;
  final bool isActive;
  final DateTime createdAt;
  final DateTime? updatedAt;

  const Rent({
    this.id = 0,
    this.propertyId = 0,
    this.propertyTitle = '',
    this.userId = 0,
    this.userName = '',
    required this.startDate,
    required this.endDate,
    this.isDailyRental = false,
    this.totalPrice = 0.0,
    this.rentStatusId = 0,
    this.rentStatusName = '',
    this.isActive = true,
    required this.createdAt,
    this.updatedAt,
  });

  factory Rent.fromJson(Map<String, dynamic> json) => _$RentFromJson(json);
  Map<String, dynamic> toJson() => _$RentToJson(this);
}
